test/Npgsql.Tests/CommandTests.cs

```diff
@@ -265,27 +265,23 @@ public async Task Cancel()
         }
 
         [Test, Description("Cancels an async query with the cancellation token")]
-        [Ignore("Flaky on CoreCLR")]
         public async Task CancelAsync()
         {
-            var cancellationSource = new CancellationTokenSource();
+            if (IsMultiplexing)
+                Assert.Ignore("Async cancellation isn't supported with multiplexing yet.");
+
             using (var conn = await OpenConnectionAsync())
-            using (var cmd = CreateSleepCommand(conn))
+            using (var cmd = CreateSleepCommand(conn, 5))
             {
-                // ReSharper disable once MethodSupportsCancellation
-                _ = Task.Factory.StartNew(() =>
-                {
-                    Thread.Sleep(300);
-                    cancellationSource.Cancel();
-                });
+                var cancellationSource = new CancellationTokenSource(300);
                 var t = cmd.ExecuteNonQueryAsync(cancellationSource.Token);
-                Assert.That(async () => await t.ConfigureAwait(false), Throws.Exception.TypeOf<NpgsqlException>());
+                Assert.That(async () => await t.ConfigureAwait(false), Throws.Exception.TypeOf<PostgresException>());
```

Comment:
There is actually an issue about this #1146.

Not sure, if we should change anything for sync cancellation - the result of the cancellation via Cancel should always be a PostgresException. It's just for the cancellation via the cancellation token, we would hide the original exception behind the OperationCanceledException.

Are we doing this as a part of this pr, or a separate one?

Reply:
> There is actually an issue about this #1146.

Of course, forgot about that.

> Not sure, if we should change anything for sync cancellation - the result of the cancellation via Cancel should always be a PostgresException

I'm not sure... I think making (sync) cancellation throw OperationCanceledException is similar to making the timeout throw TimeoutException - in both cases we're translating a lower-level exception type to a standard, higher-level .NET exception that conceptually means what we want (cancellation, timeout). Do you have any specific argument on why this should continue throwing PostgresException, apart from it being a breaking change (which is definitely an argument)?

> It's just for the cancellation via the cancellation token, we would hide the original exception behind the OperationCanceledException.

It's definitely possible, though I don't think I've ever seen an inner exception on a OperationCanceledException. It could provide the extra information on where exactly the operation was cancelled (assuming a big stack with different components, and a cancellation token being passed around to everyone).

> Are we doing this as a part of this pr, or a separate one?

As you're already working right in the area, it could make sense to just do it - but it's totally up to you. I'm fine with merging without this and fixing later.

Reply:
BTW another small argument for throwing OperationCanceledException for sync cancellation, is that it would make sync and async more consistent.